src/workerd/api/queue.c++

```diff
@@ -294,6 +329,12 @@ kj::Promise<void> WorkerQueue::sendBatch(jsg::Lock& js, jsg::Sequence<MessageSen
   headers.add("CF-Queue-Largest-Msg"_kj, kj::str(largestMessage));
   headers.set(kj::HttpHeaderId::CONTENT_TYPE, MimeType::JSON.toString());
 
+  KJ_IF_SOME(opts, options) {
+    KJ_IF_SOME(secs, opts.delaySecs) {
```

Comment:
1. Are we ever validating that this is beneath the max delay?
2. Pending the result of the question around whether we want to support this option both on individual messages within a batch and within a batch, what happens if the user specifies a delay for the entire batch *and* for messages within the batch? Is that meant to be allowed?

Reply:
FWIW I'm fine with just doing the validation in the queue service rather than here. Feel free to rip it out of here. It just didn't look right to do some validation here but to miss some cases.